### PR TITLE
Add initial implementation of BAR visualization

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

**Comment by ChadNedzlek** [comment]
Probably shouldn't/don't need to check in the angular shield. :-)

File: src/Maestro/maestro-angular/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Editor configuration, see https://editorconfig.org
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 2
```
Comment (ChadNedzlek) [inline src/Maestro/maestro-angular/.editorconfig]: This 2 is... pretty narrow. And conflicts with the style of pretty much all files everywhere. C# is more indented in general than TS, and we use 4 there.

File: src/Maestro/maestro-angular/src/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+function log(str: string) {
+  // console.log(str);
+}
+
+export function topologicalSort<TNode, TKey>(nodes: TNode[], getChildren: (node: TNode) => TNode[], getKey: (node: TNode) => TKey): TNode[] {
```
Comment (ChadNedzlek) [inline src/Maestro/maestro-angular/src/helpers.ts]: This wouldn't be a terrible thing to add a test for. I... think it works, but it's a fairly complicated algorithm.

File: src/Maestro/maestro-angular/src/index.prod.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<html lang="en">
+<head>
+</head>
+<body>
+</body>
```
Comment (ChadNedzlek) [inline src/Maestro/maestro-angular/src/index.prod.html]: ? What's this file.  It's just an empty HTML file.

File: src/Maestro/maestro-angular/src/styles.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/* You can add global styles to this file, and also import other style files */
+
+@import "~bootstrap/scss/bootstrap";
```
Comment (ChadNedzlek) [inline src/Maestro/maestro-angular/src/styles.scss]: Does angular do CSS shaking?  We use a very, very small subset of bootstrap, so we could have a much smaller footprint if we only included the like 4 styles we actually care about (we don't have any forms, for example, which are a big pile of CSS)

File: src/Maestro/maestro-angular/src/helpers.ts (hunk 2 above)
Comment (alexperovich) [reply src/Maestro/maestro-angular/src/helpers.ts]: ill add some.

File: src/Maestro/maestro-angular/src/index.prod.html (hunk 3 above)
Comment (alexperovich) [reply src/Maestro/maestro-angular/src/index.prod.html]: This is an empty file to let angular inject its styles and js into. The C# thing can then extract the injected stuff easily.

File: src/Maestro/maestro-angular/src/app/widget/side-bar-channel/side-bar-channel.component.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  public trimName = prettyRepository;
+
+  private routeSubscription?: SubscriptionLike;
+
+  public ngOnInit() {
+    this.branches$ = this.maestro.defaultChannels.listAsync(undefined, this.channel.id);
+
+    // Get the current route state, and concat it with any changes
+    this.routeSubscription =
+      of(this.router.routerState.root).pipe(
```
Comment (ChadNedzlek) [inline src/Maestro/maestro-angular/src/app/widget/side-bar-channel/side-bar-channel.component.ts]: >of(this.router.routerState.root).pipe( [](start = 6, length = 38)

But... of(foo).pipe(stuff).subscript(bar) is just... bar(stuff(foo)).

Why make an observable of a fixed value with of?  My understanding of rxjs is that it doesn't do anything.

Comment (alexperovich) [reply src/Maestro/maestro-angular/src/app/widget/side-bar-channel/side-bar-channel.component.ts]: This is to take the current value, then concat it with the new value when a NavigationEnd happens. this of(stuff).pipe(concat(stuff)) == emit the value of `stuff` once now, then again whenever navigation end happens.

I think the of().pipe syntax is nicer than concat(<multiline thing>)(of(stuff))

File: src/Maestro/maestro-angular/.editorconfig (hunk 1 above)
Comment (alexperovich) [reply src/Maestro/maestro-angular/.editorconfig]: This is angular default, and all the templates generate files with this indent_size. I'm not going to fight with it.

**Pull request merged**